lib/tasks/delete_dupes.rake

```diff
@@ -0,0 +1,73 @@
+namespace :songs do
+  desc 'Song: Attempt to delete duplicate song sheets in a directory'
+  task delete_dupes: :environment do
+    VALID_OPTS = ['dir=DIR_NAME', 'recursive=[true|false]']
+
+    options = OpenStruct.new
+    options.dir = ENV['dir'] || ''
+    options.recursive = ENV['recursive'] == 'true'
+
+    if options.dir.blank?
+      puts "Options are: #{VALID_OPTS.join(' ')}"
+      exit
+    else
+      options.dir = File.expand_path(options.dir)
+    end
+
+    new_dir = "#{options.dir}.new"
+    FileUtils.copy_entry(options.dir, new_dir, false, false, true)
+
+    deleted_files = Set[]
+    search_str = "#{new_dir}/*"
+    search_str << '*' if options.recursive
+    Dir.glob(search_str).each do |file|
+      next if File.directory?(file)
+      next if deleted_files.include?(file)
+
+      base_name = File.basename(file, File.extname(file))
+      next if base_name.split.length < 3 # not enough words to go by
+
+      puts "Looking at '#{base_name}'...".white
+
+      similar_files = Dir.glob("#{new_dir}/#{base_name}*")
+      similar_files.delete(file) # don't want to remove the original
+
+      if similar_files.any?
+        puts "\nremoving: \n\t#{similar_files.join("\n\t")}\n".red
+        FileUtils.rm(similar_files)
+        deleted_files += similar_files
+      elsif file.match(/ \(.*\)(?=\.)| in \w{1,2}(?=\.)/)
+        # matches againt 'whatever (B).doc' or 'whatever in B.'.
+
+        # It's possible that dupes may exist as
+        # 'same file (A).doc', 'same file in A.doc'. Using Dir.glob with the
+        # whole filename as the prefix to search against won't capture these.
+        search = base_name.split[0..-2].join(' ') # remove the end piece (A)
```

Comment:
Might want to refactor some of this into some methods? Similar to what you did with `parse_songs_in_dir` in `import.rake`. Good hint that you need to move complex logic into it's own method is if you're writing comments to explain a bunch. Put it in a method with a good descriptive name, and no need to comment any more (or at least minimize). Even if the code isn't reused, I can read the `import` task much easier than this one because I understand what `parse_songs_in_dir` means whereas I have to read a bunch here.